File: Backend/ClearChain.Domain/Entities/Message.cs
namespace ClearChain.Domain.Entities;

public class Message
{
    public Guid Id { get; set; }
    public Guid PickupRequestId { get; set; }
    public Guid SenderId { get; set; }
    public Guid ReceiverId { get; set; }
    public string Content { get; set; } = string.Empty;
    public bool IsRead { get; set; } = false;
    public DateTime SentAt { get; set; }
    public DateTime? ReadAt { get; set; }

    public Organization? Sender { get; set; }
    public Organization? Receiver { get; set; }
    public PickupRequest? PickupRequest { get; set; }
}

File: Mobile/app/src/main/java/com/clearchain/app/domain/model/Message.kt
package com.clearchain.app.domain.model

data class Message(
    val id: String,
    val pickupRequestId: String,
    val senderId: String,
    val senderName: String,
    val receiverId: String,
    val content: String,
    val isRead: Boolean,
    val sentAt: String,
    val readAt: String?
)

data class MessageThread(
    val pickupRequestId: String,
    val lastMessage: Message,
    val unreadCount: Int
)
